app/src/marketplace/components/Modal/WhitelistEditProgressDialog.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { useMemo } from 'react'
+import { I18n } from 'react-redux-i18n'
+import styled from 'styled-components'
+
+import ModalPortal from '$shared/components/ModalPortal'
+import Dialog from '$shared/components/Dialog'
+import ProgressBar from '$shared/components/ProgressBar'
+import { transactionStates } from '$shared/utils/constants'
+
+import PendingTasks from './PendingTasks'
+
+const PublishProgress = styled.div`
+    width: 100%;
+`
+
+const UnstyledWhitelistEditProgressDialog = ({ onCancel, status, isPrompted, ...props }) => {
+    const { pending, progress } = useMemo(() => Object.keys(status).reduce((result, key) => {
+        const value = status[key]
+
+        if (value === transactionStates.PENDING) {
+            return {
+                ...result,
+                pending: [
+                    ...result.pending,
+                    key,
+                ],
+                progress: result.progress + 1,
+            }
+        }
+
+        if (value === transactionStates.FAILED || value === transactionStates.CONFIRMED) {
+            return {
+                ...result,
+                progress: result.progress + 2,
+            }
+        }
+
+        return result
+    }, {
+        pending: [],
+        progress: 0,
+    }), [status])
+
+    return (
+        <ModalPortal>
+            <Dialog
+                onClose={onCancel}
+                title={I18n.t('modal.whiteListEdit.title')}
+                actions={{
+                    cancel: {
+                        title: I18n.t('modal.common.cancel'),
+                        onClick: () => onCancel(),
+                        kind: 'link',
+                        disabled: true,
+                    },
+                    close: {
+                        title: I18n.t('modal.common.working'),
+                        kind: 'primary',
+                        disabled: true,
+                        onClick: () => onCancel(),
+                    },
+                }}
+                {...props}
+            >
+                <PublishProgress>
+                    <PendingTasks isPrompted={isPrompted}>
+                        {pending && pending.length > 0 && pending.map((key) => (
+                            I18n.t(`modal.whiteListEdit.actions.${key}.pending`)
+                        )).join(', ')}
+                    </PendingTasks>
+                    <ProgressBar value={((progress + 1) / ((Object.keys(status).length * 2) + 1)) * 100} />
```

Comment:
@juhah can you do one extra thing for us? `ProgressBar` uses `value` in a `styled-component` way, see [`ProgressBar/index.jsx#L17`](https://github.com/streamr-dev/streamr-platform/blob/development/app/src/shared/components/ProgressBar/index.jsx#L17). It's wrong. For each `value` a new CSS class is generated. It does `ceil` them values but it is at worst 100 classes.

Can you move that logic into `style` attribute? I.e. `style={{ width: … }}`

Reply:
Can you briefly explain the logic of calculating and displaying progress? Why `progress + 1` and why `(.length * 2) + 1`? I wanna understand it, that's all! 😃

Reply:
Let's see if I remember correctly... 

> Why progress + 1 

It starts with an initial value so it looks like something has started (don't start progress from 0)

> why (.length * 2) + 1

Each transaction has two "slots", pending advances it one step and then another completes it (with error / success). Last +1 compensates for the initial value